package com.investis.testing;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.util.Properties;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.openqa.selenium.edge.EdgeDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.safari.SafariDriver;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

public class Base {

	WebDriver driver;
	public String url="";
	public Properties prop;
	
	public void loadPropertiesFile() {
        String path = "C://Users//Dell Enterprise//eclipse-workspace//Testing//src//test//java//Utilities//config.properties";
        prop = new Properties();
        File propFile = new File(path); // Corrected line

        try (FileInputStream fis = new FileInputStream(propFile)) {
            prop.load(fis);
          //  System.out.println("Defult url=" + prop.getProperty("url"));
        } catch (IOException e) { // Catching a more specific exception
            e.printStackTrace();
            System.err.println("Error loading properties file: " + e.getMessage());
        }
    }
	public WebDriver intilizeBrowserAndOpenApplication(String browser) {
		if(browser.equalsIgnoreCase("chrome")) 
			{
	//		System.setProperty("webdriver.chrome.driver", "C:/Program Files/Google/Chrome/Application/chrome.exe");
			 // Create ChromeOptions and add incognito argument
			ChromeOptions options = new ChromeOptions();
			options.addArguments("--disable-dev-shm-usage");
			options.addArguments("--disable-extensions");
	        options.addArguments("--incognito");
			driver = new ChromeDriver(options);
			
			}
		else if(browser.equalsIgnoreCase("firefox"))
			driver = new FirefoxDriver();
		else if(browser.equalsIgnoreCase("edge"))
			driver = new EdgeDriver();
		else if(browser.equalsIgnoreCase("safari"))
			driver = new SafariDriver();
		else {
			System.out.println("valid browser is not provided, hence quitting the automation run");
			System.exit(0);
		}
		System.out.println("Browser launched and navigated to site=");	
		//url=prop.getProperty("url");

		if(url!="")
			driver.get(url);
		else
			driver.get("about:blank");
		return driver;
	}
}
